src/features/TakeNow/ui/ValidateTakeNowParams.tsx

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useRef, useState } from 'react';
+
+import { useActivityByIdQuery } from '~/entities/activity';
+import { appletModel, useAppletByIdQuery } from '~/entities/applet';
+import { useSubjectQuery } from '~/entities/subject';
+import { useUserState } from '~/entities/user/model/hooks';
+import { useWorkspaceAppletRespondent, useWorkspaceRolesQuery } from '~/entities/workspace';
+import { WorkspaceRole } from '~/shared/api/types/workspace';
+import { useNotification } from '~/shared/ui';
+import Loader from '~/shared/ui/Loader';
+import { useCustomTranslation } from '~/shared/utils';
+import { ActivityGroups } from '~/widgets/ActivityGroups';
+
+type ValidateTakeNowParamsProps = {
+  appletId: string;
+  startActivityOrFlow: string;
+  subjectId: string;
+  respondentId: string;
+};
+
+const TAKE_NOW_ROLES: WorkspaceRole[] = ['super_admin', 'owner', 'manager'];
+
+function ValidateTakeNowParams({
+  appletId,
+  subjectId: targetSubjectId,
+  startActivityOrFlow,
+  respondentId,
+}: ValidateTakeNowParamsProps) {
+  const [workspaceId, setWorkspaceId] = useState<string>('');
+
+  const { isError: isSubjectError, isLoading: isLoadingSubject } = useSubjectQuery(targetSubjectId);
+
+  const {
+    isError: isAppletError,
+    data: appletData,
+    isLoading: isLoadingApplet,
+  } = useAppletByIdQuery({ isPublic: false, appletId });
+
+  const { isError: isActivityError, isLoading: isLoadingActivity } = useActivityByIdQuery({
+    isPublic: false,
+    activityId: startActivityOrFlow,
+  });
+
+  const {
+    isError: isWorkspaceRolesError,
+    data: workspaceRolesData,
+    isLoading: isLoadingWorkspaceRoles,
+  } = useWorkspaceRolesQuery(workspaceId, {
+    appletIds: [appletId],
+  });
+
+  const {
+    isError: isRespondentError,
+    data: respondentData,
+    isLoading: isLoadingRespondent,
+  } = useWorkspaceAppletRespondent(workspaceId, appletId, respondentId);
+
+  const { showErrorNotification: showError } = useNotification();
+  const { t } = useCustomTranslation();
+  const { user } = useUserState();
+  const { initiateTakeNow, getMultiInformantState } = appletModel.hooks.useMultiInformantState();
+  const errorNotificationRef = useRef<number>(0);
+
+  const ActivityList = () => <ActivityGroups isPublic={false} appletId={appletId} />;
+
+  // This is a temporary hack to prevent showing many error notifications
+  // Working on a more permanent solution
+  const showErrorNotification = useCallback(
+    (message: string) => {
+      if (errorNotificationRef.current < 2) {
+        showError(message);
+        errorNotificationRef.current += 1;
+      }
+    },
+    [showError, errorNotificationRef],
+  );
```

Comment:
The notifications don't seem to handle multiple calls well. So far in the app they are only called in response to a user action, but when used as a side effect it renders multiple notifications.

Working on a fix for this

Reply:
I've implemented a fix for it in #440, so I'm gonna revert this workaround